flashflood/__init__.py

```diff
@@ -52,22 +52,91 @@ def __init__(self, s3_resource, bucket, root_prefix):
         self._journal_pfx = f"{root_prefix}/journals"
         self._blobs_pfx = f"{root_prefix}/blobs"
         self._new_pfx = f"{root_prefix}/new"
+        self._update_pfx = f"{root_prefix}/update"
         self._index_pfx = f"{root_prefix}/index"
 
     def put(self, data, event_id: str=None, date: datetime=None):
+        """
+        Write new event. If event exists, write update marker for event.
+        """
         date = date or datetime.utcnow()
         timestamp = datetime_to_timestamp(date)
         event_id = event_id or str(uuid4())
         assert _JournalID.DELIMITER not in event_id
-        blob_id = str(uuid4())
-        journal_id = _JournalID.make(timestamp, None, blob_id)
-        manifest = dict(journal_id=journal_id,
-                        from_date=timestamp,
-                        to_date=timestamp,
-                        size=len(data),
-                        events=[dict(event_id=event_id, timestamp=timestamp, offset=0, size=len(data))])
-        self._upload_journal(_Journal(journal_id, manifest, io.BytesIO(data)), is_new=True)
-        return Event(event_id, date, data)
+        if self.event_exists(event_id):
+            self._upload_update_action(event_id, "update", data)
+            return Event(event_id, None, data)
+        else:
+            blob_id = str(uuid4())
+            journal_id = _JournalID.make(timestamp, None, blob_id)
+            manifest = dict(journal_id=journal_id,
+                            from_date=timestamp,
+                            to_date=timestamp,
+                            size=len(data),
+                            events=[dict(event_id=event_id, timestamp=timestamp, offset=0, size=len(data))])
+            self._upload_journal(_Journal(journal_id, manifest, io.BytesIO(data)), is_new=True)
+            return Event(event_id, date, data)
+
+    def delete(self, event_id):
```

Comment:
missing typing

Reply:
Type are needed throughout this repo. That will be addressed in a future PR.